test_runner/regress/test_evttrig_extension.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import shutil
+from logging import info
+from pathlib import Path
+from typing import TYPE_CHECKING, cast
+
+import pytest
+from fixtures.log_helper import log
+from fixtures.paths import BASE_DIR
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from fixtures.neon_fixtures import (
+        NeonEnvBuilder,
+    )
+    from fixtures.pg_version import PgVersion
+
+
+# use neon_env_builder_local fixture to override the default neon_env_builder fixture
+# and use a test-specific pg_install instead of shared one
+@pytest.fixture(scope="function")
+def neon_env_builder_local(
+    neon_env_builder: NeonEnvBuilder,
+    test_output_dir: Path,
+    pg_distrib_dir: Path,
+    pg_version: PgVersion,
+) -> NeonEnvBuilder:
+    test_local_pginstall = test_output_dir / "pg_install"
+    log.info(f"copy {pg_distrib_dir} to {test_local_pginstall}")
+
+    # We can't copy only the version that we are currently testing because other
+    # binaries like the storage controller need specific Postgres versions.
+    shutil.copytree(pg_distrib_dir, test_local_pginstall)
+
+    # Now copy the SQL only extension test_evttrig_extension in the local
+    # pginstall extension directory on-disk
+    test_evttrig_extension_dir = (
+        BASE_DIR / "test_runner" / "regress" / "data" / "test_evttrig_extension"
+    )
```

Comment:
I think we should copy the extension files into the install tree in the test. I'd like to see this fixture moved to a common place and shared with `test_remote_extension`.

Reply:
Done now, after some chat to help me understand the share-code opportunity here.